### ISSUE TITLE
Attempt to read property "total" on string (View: vendor/getcandy/admin/resources/views/livewire/pages/orders/show.blade.php)

### ISSUE THREAD

**Issue by jishadp** [getcandy/getcandy#165]
I have created an order to getCandy database.

     $order = Order::create([
            'user_id'   => 1,
            'channel_id' => 2,
            'sub_total' => 21,
            'status'    =>'confirmed',
            'tax_breakdown' =>['name'=>'VAT','rate'=>3,'amount'=>12],
            'tax_total' => 0,
            'total' => 22,
            'currency_code' => 1
        ]);
        foreach ($cartLines as $cart) {
            $line = OrderLine::create([
                'order_id'  => $order->id,
                'purchasable_type'  =>$cart->purchasable_type,
                'purchasable_id'    => $cart->purchasable_id,
                'description'   =>'test',
                'identifier'    => 'dff',
                'type'  =>'physical',
                'unit_price'    =>25,
                'unit_quantity' => 1,
                'quantity' => $cart->quantity,
                'sub_total' => 523,
                'tax_breakdown' =>['name'=>'VAT','rate'=>3,'amount'=>12],
                'tax_total' => 0,
                'total' => 523
            ]);
        }

        $orderAddress = new OrderAddress;
        $orderAddress->order_id = $order->id;
        $orderAddress->country_id = 251;
        $orderAddress->title = null;
        $orderAddress->first_name = 'Jacbob';
        $orderAddress->last_name = null;
        $orderAddress->company_name = null;
        $orderAddress->line_one = '123 Foo Street';
        $orderAddress->line_two = null;
        $orderAddress->line_three = null;
        $orderAddress->city = 'London';
        $orderAddress->state = null;
        $orderAddress->postcode = 'NW1 1WN';
        $orderAddress->delivery_instructions = null;
        $orderAddress->contact_email = null;
        
        $orderAddress->contact_phone = null;
        $orderAddress->type = 'shipping'; // billing/shipping
        $orderAddress->shipping_option = null; // A unique code for you to identify shipping
        $orderAddress->save();
        
        $orderAddress = new OrderAddress;
        $orderAddress->order_id = $order->id;
        $orderAddress->country_id = 251;
        $orderAddress->title = null;
        $orderAddress->first_name = 'Jacbob';
        $orderAddress->last_name = null;
        $orderAddress->company_name = null;
        $orderAddress->line_one = '123 Foo Street';
        $orderAddress->line_two = null;
        $orderAddress->line_three = null;
        $orderAddress->city = 'London';
        $orderAddress->state = null;
        $orderAddress->postcode = 'NW1 1WN';
        $orderAddress->delivery_instructions = null;
        $orderAddress->contact_email = null;
        
        $orderAddress->contact_phone = null;
        $orderAddress->type = 'billing'; // billing/shipping
        $orderAddress->shipping_option = null; // A unique code for you to identify shipping
        $orderAddress->save();
        
        Transaction::create([
            'order_id'  => $order->id,
            'success'   => 1,
            'driver'    =>'stripe',
            'amount'    =>22,
            'reference' => 'SSKJUTREWKW',
            'status'    =>'settled',
            'card_type' =>'visa',
            'last_four' =>1344,
        ]);

When I try to access from admin hub, Its showing error like this

Attempt to read property "total" on string (View: vendor/getcandy/admin/resources/views/livewire/pages/orders/show.blade.php)

**Comment by glennjacobs** [comment]
I'll ask @alecritson to take a look at this tomorrow when he's back in.

**Comment by alecritson** [comment]
Looks like there could be an issue with this line:

```php
'tax_breakdown' =>['name'=>'VAT','rate'=>3,'amount'=>12],
```

Try this instead:

```php
[{"name": "VAT", "total": 123, "percentage": 20}]
```

**Comment by jishadp** [comment]
@alecritson I tried the below comment. 

[https://drive.google.com/file/d/1_0NpxOoEX_jK8-kYH25mrbg8EeJ3_Wfq/view?usp=sharing](url)

I  saved to table like below and still its showing the total error in hub. Please have a look at the google drive image of my table value.

Thanks

**Comment by alecritson** [comment]
Did you also change it on the orders table?

**Comment by jishadp** [comment]
Yes. 

Let me paste my code here..

       $taxBreakdown = ["name"=> "VAT", "total"=> 123, "percentage"=> 20];
        $order = Order::create([
            'user_id'   => 1,
            'channel_id' => 1,
            'sub_total' => 21,
            'status'    =>'confirmed',
            'tax_breakdown' =>$taxBreakdown,
            'tax_total' => 0,
            'total' => 22,
            'currency_code' => 1
        ]);
        foreach ($cartLines as $cart) {
            $line = OrderLine::create([
                'order_id'  => $order->id,
                'purchasable_type'  =>$cart->purchasable_type,
                'purchasable_id'    => $cart->purchasable_id,
                'description'   =>'test',
                'identifier'    => 'dff',
                'type'  =>'physical',
                'unit_price'    =>25,
                'unit_quantity' => 1,
                'quantity' => $cart->quantity,
                'sub_total' => 523,
                'tax_breakdown' =>$taxBreakdown,
                'tax_total' => 0,
                'total' => 523
            ]);
        }

**Comment by alecritson** [comment]
If you do:

```php
$cart->getManager()->createOrder();
```

Instead of creating everything manually, do you still get the same issue? It's hard to where things might be going wrong as I can't see how everything is set up

**Comment by jishadp** [comment]
I tried with $cart->getManager()->createOrder();

Call to a member function count() on null

Getting error like the  image. I think my user and customer linking is not correct. Please assist me how can I solve this ?

[https://drive.google.com/file/d/1MwMoYrtxIlaSoBuM2DmU78uP6_FZetEP/view?usp=sharing](url)

**Comment by glennjacobs** [comment]
I don't believe this is a bug, so moving to a discussion. If it does turn out to be a bug, please submit a new issue. Thanks.
